assets/js/components/EntityHeader.js

```diff
@@ -79,12 +80,17 @@ const EntityHeader = () => {
 			const maxChars = ( availableWidth * 2 ) / fontSize;
 
 			if ( maxChars < entityURL.length ) {
-				const extraChars = entityURL.length - maxChars + 4; // 4 is the length of "/...".
 				const shortenedURL = new URL( entityURL );
+				// We will further shorten the URL below by removing protocol/www (normalizing).
+				const extraChars =
+					normalizeURL( shortenedURL.toString() ).length -
+					maxChars +
+					4; // 4 is the length of "/...".
 				const origin = shortenedURL.origin;
 				const restOfURL = shortenedURL.toString().replace( origin, '' );
 				const newRestOfURL = '/...' + restOfURL.substr( extraChars );
-				setURL( origin + newRestOfURL );
+
+				setURL( normalizeURL( origin ) + newRestOfURL );
```

Comment:
I think it would be cleaner to move this into a utility function like `shortenURL( url, maxChars )`, including Jest tests for it.

Also, pending the response on https://github.com/google/site-kit-wp/issues/4428#issuecomment-1010995527, we may want to strip the protocol and domain entirely at first and only shorten further if it then _still_ doesn't fit.